src/containerapp/azext_containerapp/containerapp_env_telemetry_decorator.py

```diff
@@ -151,3 +152,172 @@ def update(self):
+        super().__init__(cmd, client, raw_parameters, models)
+        self.managed_env_def = {}
+        self.existing_managed_env_def = {}
+
+    def get_argument_otlp_name(self):
+        return self.get_param("otlp_name")
+    
+    def get_argument_endpoint(self):
+        return self.get_param("endpoint")
+    
+    def get_argument_insecure(self):
+        return self.get_param("insecure")
+    
+    def get_argument_headers(self):
+        return self.get_param("headers")
+
+    def get_argument_enable_open_telemetry_traces(self):
+        return self.get_param("enable_open_telemetry_traces")
+    
+    def get_argument_enable_open_telemetry_logs(self):
+        return self.get_param("enable_open_telemetry_logs")
+    
+    def get_argument_enable_open_telemetry_metrics(self):
+        return self.get_param("enable_open_telemetry_metrics")
+
+    def construct_payload(self):
+        # Get current containerapp env telemetry properties
+        try:
+            self.existing_managed_env_def = self.client.show(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(), name=self.get_argument_name())
+        except Exception as e:
+            handle_raw_exception(e)
+
+        self.set_up_otlp()
+        self.set_up_enable_open_telemetry_traces()
+        self.set_up_enable_open_telemetry_logs()
+        self.set_up_enable_open_telemetry_metrics()
+
+    def construct_remove_payload(self):
+        # Get current containerapp env telemetry properties
+        try:
+            self.existing_managed_env_def = self.client.show(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(), name=self.get_argument_name())
+        except Exception as e:
+            handle_raw_exception(e)
+
+        otlp_name = self.get_argument_otlp_name()
+        existing_otlps = safe_get(self.existing_managed_env_def, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "otlpConfigurations")
+        if existing_otlps is not None:
+            otlp = [p for p in existing_otlps if p["name"].lower() == otlp_name.lower()]
+            if otlp is not None:
+                idx = [i for i, p in enumerate(existing_otlps) if p["name"].lower() == otlp_name.lower()][0]
+                otlp_to_remove = existing_otlps[idx]
+                existing_otlps.remove(otlp_to_remove)
+                safe_set(self.managed_env_def, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "otlpConfigurations", value=existing_otlps)
+            else:
+                raise ValidationError(f"No otlp entry with --otlp-name {otlp_name} found")
+        else:
+            raise ValidationError(f"No otlp entry with --otlp-name {otlp_name} found")
+        
+        self.set_up_enable_open_telemetry_traces()
+        self.set_up_enable_open_telemetry_logs()
+        self.set_up_enable_open_telemetry_metrics()
+        
+    def set_up_otlp(self):
```

Comment:
>set_up_otlp [](http://example.com/codeflow?start=8&length=11)

what does set_up_otlp mean? is it contruct_payload?

Reply:
contruct the otlp section